lib/sms/sms_submit.c

```diff
@@ -88,6 +88,32 @@ static int sms_submit_encode_number(
 	return 0;
 }
 
+/**
+ * @brief Prints error information for positive error codes of @c nrf_modem_at_printf.
+ *
+ * @param[in] err Error code.
+ */
+static void sms_submit_print_error(int err)
+{
+#if (CONFIG_SMS_LOG_LEVEL >= LOG_LEVEL_ERR)
+	if (err <= 0) {
+		return;
+	}
+
+	switch (nrf_modem_at_err_type(err)) {
+	case NRF_MODEM_AT_ERROR:
+		LOG_ERR("ERROR: %d", nrf_modem_at_err(err));
```

Comment:
`nrf_modem_at_err()` is intended to be used only with CME and CMS errors. `NRF_MODEM_AT_ERROR` means that modem responded with `ERROR` and there's no more detailed information available.

Reply:
True, fixed.